201-timeseriesinsights-environment-with-iothub/azuredeploy.json

```diff
@@ -0,0 +1,252 @@
+      "apiVersion": "2016-02-03",
+      "type": "Microsoft.Devices/IotHubs",
+      "name": "[parameters('iotHubName')]",
+      "location": "[parameters('location')]",
+      "sku": {
+        "name": "[parameters('iotHubSku')]",
+        "tier": "[parameters('iotHubSkuTier')]",
+        "capacity": 1
+      }
+    },
+    {
+      "condition": "[empty(parameters('existingIoTHubResourceId'))]",
+      "apiVersion": "2016-02-03",
+      "name": "[concat(parameters('iotHubName'), '/events/', parameters('consumerGroupName'))]",
+      "type": "Microsoft.Devices/iotHubs/eventhubEndpoints/ConsumerGroups",
+      "dependsOn": [
+        "[concat('Microsoft.Devices/iothubs/', parameters('iotHubName'))]"
+      ]
+    },
+    {
+      "type": "Microsoft.TimeSeriesInsights/environments",
+      "name": "[parameters('environmentName')]",
+      "apiVersion": "2017-11-15",
+      "location": "[parameters('location')]",
+      "properties": {
+        "dataRetentionTime": "[parameters('environmentDataRetentionTime')]"
+      },
+      "sku": {
+        "name": "[parameters('environmentSkuName')]",
+        "capacity": "[parameters('environmentSkuCapacity')]"
+      },
+      "tags": "[variables('environmentTagsValue')]",
+      "resources": [
+        {
+          "type": "eventsources",
+          "name": "[parameters('eventSourceName')]",
+          "apiVersion": "2017-11-15",
+          "location": "[parameters('location')]",
+          "kind": "Microsoft.IoTHub",
+          "properties": {
+            "eventSourceResourceId": "[variables('eventSourceResourceId')]",
+            "iotHubName": "[parameters('iotHubName')]",
+            "consumerGroupName": "[parameters('consumerGroupName')]",
+            "keyName": "[parameters('eventSourceKeyName')]",
+            "sharedAccessKey": "[listkeys(concat(variables('eventSourceResourceId'), '/IoTHubKeys/', parameters('eventSourceKeyName')), '2016-02-03').primaryKey]",
+            "timestampPropertyName": "[parameters('eventSourceTimestampPropertyName')]"
+          },
+          "tags": "[variables('eventSourceTagsValue')]",
+          "dependsOn": [
+            "[concat('Microsoft.TimeSeriesInsights/environments/', parameters('environmentName'))]",
+            "[concat('Microsoft.Devices/IotHubs/', parameters('iotHubName'))]",
+            "[concat('Microsoft.Devices/IotHubs/', parameters('iotHubName'), '/eventHubEndpoints/events/ConsumerGroups/', parameters('consumerGroupName'))]"
+          ]
+        }
+      ]
+    },
+    {
+      "condition": "[not(empty(parameters('accessPolicyReaderObjectIds')))]",
+      "type": "Microsoft.TimeSeriesInsights/environments/accesspolicies",
+      "name": "[concat(parameters('environmentName'), '/', 'readerAccessPolicy', copyIndex())]",
+      "copy": {
+        "name": "accessPolicyReaderCopy",
+        "count": "[if(empty(parameters('accessPolicyReaderObjectIds')), 1, length(parameters('accessPolicyReaderObjectIds')))]"
```

Comment:
have you tested this with an empty array (not sure it will work due to a "bug" in ARM)